pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -2822,6 +2832,37 @@ func checkAndPublishAppInstanceConfig(getconfigCtx *getconfigContext,
 		config.Errors = append(config.Errors, err.Error())
 	}
 
+	// If EnforceNetworkInterfaceOrder is enabled, check that every network interface
+	// has unique order number.
```

Comment:
Would it be possible to add some extra validation at this stage? I think it would be useful to report if the config provided cannot be "executed," for example, taking into account the restrictions of the multifunc PCI device (it cannot separate interfaces). But I'm not sure we have enough info at this stage...

Reply:
I'm not sure if this is the right place since for other hypervisors we may implement interface ordering differently and this limitation might not apply.